src/picongpu/include/plugins/kernel/CopySpecies.kernel

```diff
@@ -33,8 +33,104 @@ namespace picongpu
 
 using namespace PMacc;
 
+/** copy particle to single frame
+ *
+ * copy particle data which are stored in linked list and superCell hierarchy
+ * to a single frame.
+ */
+struct CopySpeciesToGlobal
+{
+    DataSpace<simDim> gridDim;
+    int threads;
 
+    CopySpeciesToGlobal(const DataSpace<simDim>& gridDim, const int threads) :
+    gridDim(gridDim), threads(threads)
+    {
+
+    }
+
+    /** copy particles
+     *
+     * @param counter scalar offset in `destFrame`
+     * @param destFrame single frame were all particles are copied in
+     * @param srcBox particle box were particles are read from
+     * @param filter filter to select particles
+     * @param gpuOffset gpu localOffset or to sliding window begin
```

Comment:
proposed rename:
```
@param particleOffset can be negative for the first GPU: localDomain.offset - globalWindow.offset
```
(set in [ADIOSWriter.hpp](https://github.com/psychocoderHPC/picongpu/blob/fabf2c11755dd4f52c46c2626b84b965186fa13e/src/picongpu/include/plugins/adios/ADIOSWriter.hpp#L899-L902)